client/src/components/BudgetCards.jsx

```diff
@@ -0,0 +1,135 @@
+
+  const budgetData = formatBudgetData(budgets, currentMonth, previousMonth)
+
+  const cardData = combineBudgetTotals(budgetData, totalsSpent, currentMonth, previousMonth)
+
+  function handleEdit(e, value) {
+    setEditing(e.target.id)
+    setGoal(value)
+  }
+
+  function cancelEdit() {
+    setEditing()
+    setGoal()
+  }
+
+  function updateGoal(key) {
+    const updatedGoal = { category: key, amount: Number(goal).toFixed(2) }
+    setEditing()
+    setGoal()
+    setBudgets({ ...budgets, [key]: Number(goal).toFixed(2) })
+    sendUpdatedGoal(updatedGoal)
+  }
+
+  async function sendUpdatedGoal(goal) {
+    try {
+      const token = localStorage.getItem(tokenKey);
+      const res = await fetch('/api/records/budgets/update', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          'Authorization': `Bearer ${token}`
+        },
+        body: JSON.stringify(goal)
+      })
+      if (!res.ok) throw new Error(`Server error: ${res.status}`)
+    } catch (err) {
+      console.err('Could not update', err)
+    }
+  }
+
+  useEffect(() => {
+    if (Object.keys(cardData).length > 1) {
+      setLoading(false)
+    }
+  }, [cardData])
+
+  return (
+    <>
+ {!loading ?
+  <MakeCards obj={cardData} currentMonth={currentMonth} previousMonth={previousMonth} handleEdit={handleEdit} editing={editing} cancelEdit={cancelEdit} updateGoal={updateGoal} setGoal={setGoal} goal={goal} /> :
+  <h2 style={{color: 'white'}}>Loading</h2>}
+    </>
+  )
+}
+
+/**
+ * Creates an array of budget cards with the data provided from a cardData object.
+ */
+function MakeCards({ obj, handleEdit, editing, cancelEdit, updateGoal, setGoal, goal, currentMonth, previousMonth }) {
+
+  const monthNow = months[currentMonth].name;
+  const lastMonth = months[previousMonth].name;
+
+  const cards = [];
+
+  for (const [key, value] of Object.entries(obj)) {
+
+    cards.push(
+      <Col className="d-flex justify-content-start" key={key}>
+        <Card style={{ width: '18rem' }} className='mx-auto my-3 border border-dark border-1'>
+          <Card.Header className="bg-secondary text-light fs-5">{categoriesOutObj[key]}</Card.Header>
+          <ListGroup variant="flush">
+            <ListGroup.Item>
+              {editing !== key ?
+                <EditGoal handleEdit={handleEdit} category={key} value={value} /> :
+                <GoalEditInput cancelEdit={cancelEdit} category={key} value={value} setGoal={setGoal} updateGoal={updateGoal} />}
+            </ListGroup.Item>
+            <ListGroup.Item>{monthNow} spending was ${Number(value[currentMonth]).toFixed(2)}</ListGroup.Item>
+            <ListGroup.Item>{lastMonth} spending was ${Number(value[previousMonth]).toFixed(2)}</ListGroup.Item>
+          </ListGroup>
+        </Card>
+      </Col>
+    )
+  }
+  return cards;
+}
+
+
+/**
+ * The default display of a budget goal with a button to enter editing mode.
+ */
+function EditGoal({ handleEdit, category, value }) {
```

Comment:
Event handler _props_ should start with `on`. The actual event handler functions would start with `handle`.